plugin/src/main/java/org/jenkinsci/plugins/structs/describable/UninstantiatedDescribable.java

```diff
@@ -0,0 +1,167 @@
+package org.jenkinsci.plugins.structs.describable;
+
+import hudson.model.Describable;
+import org.jenkinsci.Symbol;
+
+import javax.annotation.Nullable;
+import java.io.Serializable;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.TreeMap;
+
+/**
+ * A reflection-like view of a particular {@link Describable} instance.
+ *
+ * @author Kohsuke Kawaguchi
+ */
+public class UninstantiatedDescribable implements Serializable {
+    private String symbol;
+    private String klass;
+    private final Map<String,Object> arguments;
+
+    public UninstantiatedDescribable(String symbol, String klass, Map<String, Object> arguments) {
+        this.symbol = symbol;
+        this.klass = klass;
+        this.arguments = arguments;
+    }
+
+    public UninstantiatedDescribable(Map<String, Object> arguments) {
+        this(null,null,arguments);
+    }
+
+    /**
+     * If this nested describable has a suitable {@linkplain Symbol symbol name},
+     * this method returns one.
+     */
+    public @Nullable String getSymbol() {
+        return symbol;
+    }
+
+    public void setSymbol(String symbol) {
+        this.symbol = symbol;
+    }
+
+    /**
+     * {@code $class} is an alternative means to specify the class in case there's no symbol.
+     * Can be a short name if it's contextually unambiguous, or a FQCN.
+     *
+     * <p>
+     * Either this or {@link #getSymbol()} has to return a non-null value.
```

Comment:
Then `getSymbol` should have matching Javadoc, to justify its being `@Nullable` rather than `@CheckForNull`.